validation/distributed_simulations/distributed_nonhydrostatic_turbulence.jl

```diff
@@ -15,34 +15,30 @@ using Statistics
 using Printf
 using Logging
 
+MPI.Init()
+
 Logging.global_logger(OceananigansLogger())
 
 comm = MPI.COMM_WORLD
 rank = MPI.Comm_rank(comm)
 Nranks = MPI.Comm_size(comm)
 
+
 @info "Running on rank $rank of $Nranks..."
 
 Nx = Ny = 256
 Lx = Ly = 2π
 topology = (Periodic, Periodic, Flat)
-arch = Distributed(CPU(); topology, ranks=(1, Nranks, 1), communicator=comm)
-grid = RectilinearGrid(arch; topology, size=(Nx ÷ Nranks, Ny), halo=(3, 3), x=(0, 2π), y=(0, 2π))
+arch = Distributed(CPU(); partition=Partition(1, Nranks, 1))
+grid = RectilinearGrid(arch; topology, size=(Nx, Ny), halo=(3, 3), x=(0, 2π), y=(0, 2π))
 
 @info "Built $Nranks grids:"
 @show grid
 
 model = NonhydrostaticModel(; grid, advection=WENO(), closure=ScalarDiffusivity(ν=1e-4, κ=1e-4))
 
-# This doesn't work?
-# ϵ(x, y, z) = 2rand() - 1 # ∈ (-1, 1)
-# set!(model, u=ϵ, v=ϵ)
-
-uᵢ = rand(size(grid)...)
-vᵢ = rand(size(grid)...)
-uᵢ .-= mean(uᵢ)
-vᵢ .-= mean(vᵢ)
-set!(model, u=uᵢ, v=vᵢ)
+ϵ(x, y) = 2rand() - 1 # ∈ (-1, 1)
```

Comment:
We need to differentiate the seed per core,
otherwise the initial conditions are the same in each core

```suggestion
Random.seed!((arch.local_rank +1) * 123) 
ϵ(x, y) = 2rand() - 1 # ∈ (-1, 1)
```